### PR TITLE
BUG 1798043: Optimize VF configure logic

### PR DESCRIPTION
**Pull request by pliurh**
Check whether desired state is the same as the current one before
executing.

### CODE REVIEW

File: pkg/utils/utils.go [hunk 1]
```diff
@@ -131,54 +131,66 @@ func SyncNodeState(newState *sriovnetworkv1.SriovNetworkNodeState) error {
...
 		glog.Errorf("configSriovDevice(): fail to set NumVfs for device %s: %v", iface.PciAddress, err)
 		return err
 	}
-
-	err := setSriovNumVfs(iface.PciAddress, iface.NumVfs)
-	if err != nil {
-		glog.Errorf("configSriovDevice(): fail to set NumVfs for device %s", iface.PciAddress)
-		return err
+	// set numVFs
+	if iface.NumVfs != ifaceStatus.NumVfs {
```
Comment (zshi-redhat) [inline pkg/utils/utils.go]: This is checked in `needUpdate`,  is there other reason to check it here again?

File: pkg/utils/utils.go [hunk 2]
```diff
@@ -131,54 +131,66 @@ func SyncNodeState(newState *sriovnetworkv1.SriovNetworkNodeState) error {
...
+			glog.Errorf("configSriovDevice(): fail to set NumVfs for device %s", iface.PciAddress)
+			return err
+		}
 	}
 	// set PF mtu
-	err = setNetdevMTU(iface.PciAddress, iface.Mtu)
-	if err != nil {
-		glog.Warningf("configSriovDevice(): fail to set mtu for PF %s: %v", iface.PciAddress, err)
-		return err
+	if iface.Mtu > 0 && iface.Mtu != ifaceStatus.Mtu {
```
Comment (zshi-redhat) [inline pkg/utils/utils.go]: ditto

File: pkg/utils/utils.go (hunk 1 above)
Comment (pliurh) [reply pkg/utils/utils.go]: needUpdate will return 'True' if any of the parameter has been updated. However, we need to check if one parameter shall be updated separately before executing.

**Pull request merged**